docs/style-guide.md

```diff
@@ -0,0 +1,559 @@
+
+*Right:*
+
+```js
+const a = 0;
+if (a !== '') {
+  console.log('winning');
+}
+
+```
+
+*Wrong:*
+
+```js
+const a = 0;
+if (a == '') {
+  console.log('losing');
+}
+```
+
+[comparisonoperators]: https://developer.mozilla.org/en/JavaScript/Reference/Operators/Comparison_Operators
+
+### Use descriptive conditions
+
+Any non-trivial conditions should be assigned to a descriptively named variable or function:
+
+*Right:*
+
+```js
+const isValidPassword = password.length >= 4 && /^(?=.*\d).{4,}$/.test(password);
+
+if (isValidPassword) {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+if (password.length >= 4 && /^(?=.*\d).{4,}$/.test(password)) {
+  console.log('losing');
+}
+```
+
+## Functions
+
+### Return early from functions
+
+To avoid deep nesting of if-statements, always return a function's value as early
+as possible.
+
+*Right:*
+
+```js
+function isPercentage(val) {
+  if (val < 0) {
+    return false;
+  }
+
+  if (val > 100) {
+    return false;
+  }
+
+  return true;
+}
+```
+
+*Wrong:*
+
+```js
+function isPercentage(val) {
+  if (val >= 0) {
+    if (val < 100) {
+      return true;
+    } else {
+      return false;
+    }
+  } else {
+    return false;
+  }
+}
+```
+
+Or for this particular example it may also be fine to shorten things even
+further:
+
+```js
+function isPercentage(val) {
+  const isInRange = (val >= 0 && val <= 100);
+  return isInRange;
+}
+```
+
+### Name your closures
+
+Feel free to give your closures a name. It shows that you care about them, and
+will produce better stack traces, heap and cpu profiles.
+
+*Right:*
+
+```js
+req.on('end', function onEnd() {
+  console.log('winning');
+});
+```
+
+*Wrong:*
+
+```js
+req.on('end', function() {
+  console.log('losing');
+});
+```
+
+### No nested closures
+
+Use closures, but don't nest them. Otherwise your code will become a mess.
+
+*Right:*
+
+```js
+setTimeout(function() {
+  client.connect(afterConnect);
+}, 1000);
+
+function afterConnect() {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+setTimeout(function() {
+  client.connect(function() {
+    console.log('losing');
+  });
+}, 1000);
+```
+
+
+### Method chaining
+
+One method per line should be used if you want to chain methods.
+
+You should also indent these methods so it's easier to tell they are part of the same chain.
+
+*Right:*
+
+```js
+User
+  .findOne({ name: 'foo' })
+  .populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+```
+
+*Wrong:*
+
+```js
+User
+.findOne({ name: 'foo' })
+.populate('bar')
+.exec(function(err, user) {
+  return true;
+});
+
+User.findOne({ name: 'foo' })
+  .populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+
+User.findOne({ name: 'foo' }).populate('bar')
+.exec(function(err, user) {
+  return true;
+});
+
+User.findOne({ name: 'foo' }).populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+```
+
+## Comments
+
+### Use slashes for comments
```

Comment:
What do you think about this rule? I think that we only need to add an exception for comments that are written in JSDoc format, for all other cases comments should be written only with slashes to avoid inconsistency when some write comments with slashes in one place and multi-line comments in other places.

Reply:
I agree if ESlint can distinguish between JSDoc and non-JSDoc. Could it?

Reply:
According to this rule: https://eslint.org/docs/rules/multiline-comment-style it ignores JSDoc comments
![image](https://user-images.githubusercontent.com/31747499/147146566-09df2400-fbf9-43d2-b35d-a6084f75a959.png)  
So we can add `"multiline-comment-style": ["error", "separate-lines"]` and it will be enough.